compiler/rustc_middle/src/query/mod.rs

```diff
@@ -965,6 +965,11 @@ rustc_queries! {
         desc { |tcx| "collecting attributes of `{}`", tcx.def_path_str(def_id) }
     }
 
+    /// Checks if item has `#[doc(hidden)]` attribute.
+    query is_doc_hidden(def_id: DefId) -> bool {
+        desc { |tcx| "checking if item is doc-hidden `{}`", tcx.def_path_str(def_id) }
+    }
```

Comment:
Consider *not* adding a query for this, and using a normal function instead. That will reduce the overhead of caching the result, since `is_doc_hidden` is very cheap to calculate.